### PR TITLE
enhancement(codecs): Add support for enhanced tags to native protobuf codec

### PR DESCRIPTION
**Pull request by bruceg**
This modifies the "event" protocol to _add_ support for the new enhanced tags. The existing tags field is maintained and tag values are loaded from both fields on input. I have added a compatibility note about this in the upgrade guide, but let me know if there are any other notes needed.

Ref: #14742

### CODE REVIEW

**Comment by neuronull** [comment]
LGTMy newb self.

Is the `!` in the PR title, intentional enthusiasim?

**Comment by bruceg** [comment]
The `!` is to mark a known breaking change.

**Comment by neuronull** [comment]
> The `!` is to mark a known breaking change.

TIL to read :facepalm: 

>   * `!` = OPTIONAL: signals a breaking change

File: website/content/en/highlights/2022-11-07-0-26-0-upgrade-guide.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Vector's 0.26.0 release includes **breaking changes**:
 1. [Removal of `check_fields` condition type](#check-fields-removal)
 2. [Removal of `lanes` parameter for `route` transform](#lanes-parameter-route-transform-removal)
 3. [Removal of `geoip` transform](#geo-transform-removal)
+4. [Changes to `native` Protobuf codec](#vector-native-protobuf)
```
Comment (spencergilbert) [inline website/content/en/highlights/2022-11-07-0-26-0-upgrade-guide.md:17]: I imagine we should make simultaneous changes to the native-json codec? (as part of the same release)

**Review by spencergilbert — APPROVED** [review]

**Review by lukesteensen — COMMENTED** [review]
It seems like we could reduce the incompatibility by writing a single tag to the old protobuf field. We'd still want to note that data will be reduced to a single tag when loaded by the old version, but that's always been the case and shouldn't be much of a concern.

In another release or two we could stop doing the additional write, but it might be worthwhile to keep for one release cycle.

**Comment by bruceg** [comment]
Thanks for the idea, @lukesteensen. With that change, this is no longer a breaking change, which is welcome.

**Review by tobz — APPROVED** [review]
This makes sense to me. 👍🏻

**Pull request merged**